torch/utils/_pytree.py

```diff
@@ -172,6 +176,26 @@ def _register_pytree_node(
     SUPPORTED_SERIALIZED_TYPES[typ] = serialize_node_def
     SERIALIZED_TYPE_TO_PYTHON_TYPE[type_fqn] = typ
 
+    if _register_cxx_pytree_node:
+        import torch
+
+        if torch._running_with_deploy():
+            return
+
+        try:
+            from . import _cxx_pytree
+        except ImportError:
+            pass
+        else:
+            _cxx_pytree.register_pytree_node(
```

Comment:
It feels a bit weird to me that the `_cxx_pytree.register_pytree_node` and `_pytree._register_pytree_node` have different signatures.

Reply:
Morally, we should register everything to both python and cxx pytree and skip the registrations for cxx pytree's built-in types. One idea is that:
1. we add a "_builtin" kwarg
2. "_builtin=True" only registers to Python pytree
3. we assert that "_builtin=True" is only being called from inside torch/utils/_pytree (by looking up the stack frame)

Reply:
Thanks for the suggestion and I will update the PR.

Reply:
I add a PR to ensure the two implementations provides the same public APIs with the same function signatures.

- #112485